include/sensors/core/fdc1004.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+// TODO (lc 02-16-2022) We should refactor the fixed point
+// helper functions such that they live in a shared location.
+#include "motor-control/core/utils.hpp"
+
+/*
+ * FDC1004 Capacitive Sensor
+ *
+ * Datasheet:
+ * https://www.ti.com/lit/ds/symlink/fdc1004.pdf?ts=1645556558559&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252FFDC1004
+ *
+ * The configuration of the sensor is always in single-ended measurement mode.
+ * We are currently only using 1 out of the 4 measurement channels.
+ *
+ * By default, CAPDAC is enabled. The max resolution of the capdac is 5 bits.
+ *
+ * Capacitance measurement is a total of 24 bits long. You must do consecutive
+ * reads of the MSB and LSB registers to get the data.
+ */
+
+namespace fdc1004_utils {
+static auto convert(uint32_t measurement, uint16_t accumulation,
```

Comment:
`static` here is not a good idea IMO - what it will do is force the function definition to live in every single translation unit that includes this file, even transitively. These are just plain old pure functions, just declare them without static i.e. `auto convert(...) -> sq14_15;`